microsoft-edge/devtools-guide-chromium/performance/extension.md

```diff
@@ -0,0 +1,651 @@
+
+   DevTools opens.
+
+1. In DevTools, select the **Performance** (![Performance icon](./extension-images/performance-icon.png)) tool.
+
+1. Refresh the page, to clear any selections.
+
+1. Click the **Record** (![Record icon](./extension-images/record-icon.png)) button.
+
+1. At the top of the demo page, in the **Camera** (![Camera icon](./extension-images/camera-icon.png)) dropdown menu on the left, change from **All** to **Apple iPhone 12**.
+
+1. Change the **Camera** filter back to **All**.
+
+1. Click the first photo.
+
+   The photo expands.
+
+1. In DevTools, click the **Stop** button.
+
+   The profile is displayed.
+
+1. Click **Customize and control DevTools** (![Customize and control DevTools icon](./extension-images/customize-and-control-devtools-icon.png)), and then next to **Dock location**, select **Undock into separate window** (![Undock into separate window icon](./extension-images/undock-into-separate-window-icon.png)).
+
+   DevTools is detached into its own window, so it's easier to see more data.
+
+1. Scroll down to the **Custom performance timings — Custom track** track group, and expand it.
+
+1. Expand the **Photo creation** track.
+
+1. Use the mouse and arrow keys to shift and zoom the profile, to display the custom performance measures.
+
+   This results in the following custom track entry in the performance timeline, along with its tooltip text and properties:
+
+   ![The Photo creation custom track](./extension-images/custom-track.png)
+
+   The events that are created by `performance.measure()` have a duration, but they are quite fast, so you must zoom in to see them in the profile.  Each of these events record the creation of an image, which happens when you change the filter (dropdown list) value, and when the whole gallery UI is refreshed.
+
+   The **Photo creation** events appear between the `Start filtering` and `Done filtering` performance marks that result from `performance.mark()` (covered in the next section):
+
+   ![The Start filtering mark and the Done filtering mark](./extension-images/marks-start-filtering-done-filtering.png)
+
+See also:
+* [performance.measure()](https://developer.mozilla.org/docs/Web/API/Performance/measure)
+* [performance.now()](https://developer.mozilla.org/docs/Web/API/Performance/now)
+
+
+<!-- ---------- -->
+###### Add markers to the Timings track (`performance.mark()`)
+<!-- https://developer.chrome.com/docs/devtools/performance/extension#markers -->
+
+Visually highlight specific points of interest in the timeline with custom markers that span across all tracks.
+
+For example, in [photo-gallery/gallery.js](https://github.com/MicrosoftEdge/Demos/blob/main/photo-gallery/gallery.js), an event listener handles `select` events on the demo page's `.filter` dropdown lists.  The event listener calls `performance.mark()` for the `Start filtering` event<!-- todo: correct use of "event"? --> and for the `Done filtering` event:<!-- todo: correct use of "event"? -->
```

Comment:
correct use of "event" at end of sentence 2x?
@captainbrosset

Reply:
```suggestion
For example, in [photo-gallery/gallery.js](https://github.com/MicrosoftEdge/Demos/blob/main/photo-gallery/gallery.js), an event listener handles `input` events on the demo page's filter dropdown lists.  The event listener calls `performance.mark()` before the new filter is applied and then again once the filter has been applied.
```